app/services/irs_attempts_api/tracker.rb

```diff
@@ -6,7 +6,7 @@ def track_event(event_type, metadata = {})
     end
 
     def parse_failure_reason(result)
-      return result.to_h[:error_details] || result.errors.presence
+      return result.to_h[:error_details]&.transform_values(&:keys) || result.errors.presence
     end
```

Comment:
we could probably yank every usage of this method and just inline it to `failure_reason: form_response.error_details`

Reply:
At this point, that's just more work? Seems like this mostly preserves the original intent, if we're okay to keep as-is.

Reply:
yeah I might make a second PR to do that, because it just means we have less code hanging around that deals with these structures, so it's less to worry about if we ever have to make another similar refactor